cpp/oneapi/dal/algo/kmeans/backend/gpu/train_kernel_lloyd_dense_dpc.cpp

```diff
@@ -100,60 +99,75 @@ struct train_kernel_gpu<Float, method::lloyd_dense, task::clustering> {
 
         const int64_t row_count = data.get_row_count();
         const int64_t column_count = data.get_column_count();
+        auto data_ptr = row_accessor<const Float>(data).pull({ 0, -1 });
```

Comment:
Make sure all pulls are called with queue and right alloc 

```suggestion
        auto data_ptr = row_accessor<const Float>(data).pull(q, { 0, -1 }, sycl::usm::alloc::device);
```